cli-plugins/apps/how-to-guides.hbs.md

```diff
@@ -0,0 +1,611 @@
+
+When creating a workload from local source code, these unused files can be added to the `.tanzuignore` file so there won't be an unnecesary consumption of resources and also, the workload deployment won't be triggered when one of those files changes. The file must contain itself as well for it to be ignored.
+
+The `.tanzuignore` file must be located at the root of the project so Apps CLI plug-in can read it.
+It allows to list folders (these must not end with the system separator, e.g. `/` or `\`), files and have comments which start with `#`. If the file contains files or directories that are not in the source code, they are ignored.
+
+**Example of a .tanzuignore file**
+```bash
+    .tanzuignore # must contain itself in order to be ignored
+    # This is a comment
+    this/is/a/folder/to/exclude
+
+    this-is-a-file.ext
+```
+
+<!-- ## <a id='maximize-efficiency'> Leveraging ENVs to maximize efficiency
+
+## <a id='param-yaml-usage'> --param-yaml
+
+## <a id='app-usage'> --app
+
+## <a id='type-usage'> --type
+
+## <a id='labels-anns-env'> --labels, --annotations, --env, --build-env -->
+
+## <a id='dry-run-usage'> --dry-run
+
+The main goal of `--dry-run` flag is to prepare all the steps to submit a workload to the cluster and stop before sending it, showing
+an output of the final structure of the workload.
+
+For example, when applying a workload from Git source:
+
+```bash
+tanzu apps workload apply rmq-app --git-repo https://github.com/jhvhs/rabbitmq-sample --git-branch main --service-ref "rmq=rabbitmq.com/v1beta1:RabbitmqCluster:example-rabbitmq-cluster-1" -t web --dry-run
+---
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  creationTimestamp: null
+  labels:
+    apps.tanzu.vmware.com/workload-type: web
+  name: rmq-app
+  namespace: default
+spec:
+  serviceClaims:
+  - name: rmq
+    ref:
+      apiVersion: rabbitmq.com/v1beta1
+      kind: RabbitmqCluster
+      name: example-rabbitmq-cluster-1
+  source:
+    git:
+      ref:
+        branch: main
+      url: https://github.com/jhvhs/rabbitmq-sample
+status:
+  supplyChainRef: {}
+```
+
+This will allow the user to check how it's workload would be created/updated in the cluster.
+If there would be an error when trying to apply the workload, this would be shown with the `--dry-run` flag. 
+
+```bash
+tanzu apps workload create rmq-sample-app --git-repo https://github.com/jhvhs/rabbitmq-sample --git-branch main --service-ref "rmq=rabbitmq.com/v1beta1:RabbitmqCluster:example-rabbitmq-cluster-1" -t web --dry-run
+Error: workload "default/rmq-sample-app" already exists
+```
+
+## <a id='update-strategy-usage'> --update-strategy
+
+A workload specification can be changed using Apps CLI plug-in through the `--update-strategy` flag, which checks whether the current workload in the cluster needs to be merged or replaced, defaulting if not specified as `merge`.
```

Comment:
The update-strategy flag can be used to control whether configuration properties/values passed in via `--file workload.yaml` for an existing workload are `merged` with, or completely `replace` (overwrite), existing on-cluster properties/values set for a workload.

The current default strategy is `merge`.
The impact of `merge` as default is that if/when a property is removed from the local workload.yaml and the workload.yaml is `tanzu apps workload apply`'d, the removed property **_will not_** be deleted from the on-cluster workload properties/values  configuration. 

If it's desired for the `--file workload.yaml` to completely overwrite the on-cluster workload properties/values  configuration, `--update-strategy replace` should be provided.

The default behavior is intended to prevent unintentional deletions of critical properties from existing workloads.

It has been found that this default is counter-intuitive for users and it has been deprecated as of TAP v1.4.0 and the default will be switched to `replace` in 2024.

Reply:
> *Note All tanzu apps workload apply commands will employ the merge update strategy regardless of whether the --update-strategy flag is provided.

This comment is not very clear to me, it would indicate me that no matter if I specified the flag, the value is going to keep being `merge`

I would go with something like this

> All `tanzu apps workload apply` commands will employ the `merge` update strategy if not specified otherwise by explicitly using the flag and setting its value to `replace`